server/src/main/java/com/linecorp/centraldogma/server/internal/api/DtoConverter.java

```diff
@@ -58,29 +60,38 @@ public static RepositoryDto convert(Repository repository) {
                                  repository.creationTimeMillis());
     }
 
-    public static <T> EntryDto<T> convert(Repository repository, Revision revision,
-                                          Entry<T> entry, boolean withContent) {
+    public static <T> EntryDto convert(Repository repository, Revision revision,
+                                       Entry<T> entry, boolean withContent) {
+        return convert(repository, revision, entry, withContent, null);
+    }
+
+    public static <T> EntryDto convert(Repository repository, Revision revision,
+                                       Entry<T> entry, boolean withContent, @Nullable QueryType queryType) {
         requireNonNull(entry, "entry");
         if (withContent && entry.hasContent()) {
-            return convert(repository, revision, entry.path(), entry.type(), entry.content());
+            // TODO(ks-yim): Not sure DtoConverter having business logic and it looks hacky to receive
+            //   queryType to handle JSON_PATH query for JSON5 files.
+            return convert(repository, revision, entry.path(), entry.type(),
+                           maybeJson5(entry) && queryType != QueryType.JSON_PATH ? entry.contentAsText()
+                                                                                 : entry.content());
         }
         return convert(repository, revision, entry.path(), entry.type());
     }
 
-    private static <T> EntryDto<T> convert(Repository repository, Revision revision,
-                                           String path, EntryType type) {
+    private static <T> EntryDto convert(Repository repository, Revision revision,
+                                        String path, EntryType type) {
         return convert(repository, revision, path, type, null);
     }
 
-    private static <T> EntryDto<T> convert(Repository repository, Revision revision, String path,
-                                           EntryType type, @Nullable T content) {
+    private static <T> EntryDto convert(Repository repository, Revision revision, String path,
+                                        EntryType type, @Nullable T content) {
```

Comment:
We don't need `<T>` anymore throughout this class :
```suggestion
    private static EntryDto convert(Repository repository, Revision revision, String path,
                                    EntryType type, @Nullable Object content) {
```